File: android/ch04/5ListFooterExam/app/src/main/java/com/example/a5_listfooterexam/adapter/JobAdapter.java
package com.example.a5_listfooterexam.adapter;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.a5_listfooterexam.R;
import com.example.a5_listfooterexam.model.Job;

import java.util.List;

public class JobAdapter extends ArrayAdapter<Job> {
    Activity activity;
    int resource;

    public JobAdapter(@NonNull Context context, int resource, @NonNull List<Job> objects) {
        super(context, resource, objects);
        activity = (Activity) context;
        this.resource = resource;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if(convertView == null) {
            convertView = activity.getLayoutInflater().inflate(resource, null);
        }
        Job item = getItem(position);

        if(item != null) {
            ImageView imageView = convertView.findViewById(R.id.imageView);
            TextView textView1 = convertView.findViewById(R.id.textView1);
            TextView textView2 = convertView.findViewById(R.id.textView2);

            imageView.setImageResource(item.getImage());
            textView1.setText(item.getTitle());
            textView2.setText(item.getDescription());
        }
        return convertView;
    }
}

File: android/ch04/3ListAdapterExamKotlin/app/src/main/java/com/example/a3_listadapterexamkotlin/adapter/JobAdapter.kt
package com.example.a3_listadapterexamkotlin.adapter

import android.app.Activity
import android.content.Context
import android.widget.ArrayAdapter
import com.example.a3_listadapterexamkotlin.model.Job

class JobAdapter(
    context: Context,
    resource: Int,
    objects: List<*>
) :
    ArrayAdapter<Job?>(context, resource, objects) {
    var activity: Activity
    var resource: Int

    // JobAdapter adapter = new JobAdapter(this, R.layout.list_item, list);
    init {
        activity = context as Activity
        this.resource = resource
    }
}
